composer/profiler/profiler.py

```diff
@@ -252,6 +262,7 @@ def merge_traces(self):
     def marker(
         self,
         name: str,
+        state: State,
```

Comment:
Commented on this above, but could we not require the state to be passed in on the `marker` API? I presume this would be one of the only user-facing APIs of the profiler and would like to keep the required arguments as minimal as possible.